From 5af7de7e3d4da47d9e8059a435c6c47c2260ede3 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 23:22:46 +0200
Subject: [PATCH] avformat/rtpdec_av1: Read the OBU header byte after the size
 checks

Moving the read after the checks was suggested in the report.

Hardening for R5y4vDJ078Lr (AISLE-2026-0111-00004)
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
(cherry picked from commit 5af7de7e3d4da47d9e8059a435c6c47c2260ede3)
---
 libavformat/rtpdec_av1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavformat/rtpdec_av1.c b/libavformat/rtpdec_av1.c
index f6b1048328e..a45a2f149d6 100644
--- a/libavformat/rtpdec_av1.c
+++ b/libavformat/rtpdec_av1.c
@@ -228,8 +228,6 @@ static int av1_handle_packet(AVFormatContext *ctx, PayloadContext *data,
             rem_pkt_size -= num_lebs;
             buf_ptr += num_lebs;
         }
-        // read first byte (which is the header byte only for non-fragmented elements)
-        obu_hdr = *buf_ptr;
         if (obu_size > rem_pkt_size) {
             av_log(ctx, AV_LOG_ERROR, "AV1 OBU size %u larger than remaining pkt size %d\n", obu_size, rem_pkt_size);
             return AVERROR_INVALIDDATA;
@@ -240,6 +238,8 @@ static int av1_handle_packet(AVFormatContext *ctx, PayloadContext *data,
             return AVERROR_INVALIDDATA;
         }
 
+        // read first byte (which is the header byte only for non-fragmented elements)
+        obu_hdr = *buf_ptr;
         if (!is_frag_cont) {
             uint8_t obu_type = (obu_hdr >> AV1S_OBU_TYPE) & AV1M_OBU_TYPE;
             if (obu_hdr & AV1F_OBU_FORBIDDEN) {
-- 
2.53.0

